### ISSUE TITLE
macOS: Mission Control window focus issue

### ISSUE THREAD

**Issue by git-brandon** [godotengine/godot#26877]
**Godot version:**

Godot_v3.1-rc1_osx.64

**OS/device including version:**

macOS Mojave 10.14.3

**Issue description:**

NOTE: I'm not sure if this is a macOS issue or a Godot issue, but it only seems to happen with this application.

When using Mission Control to switch back to Godot, it sometimes switches to the application, but leaves focus on the previous application window. It does switch to the correct application as the Godot menu displays in the menu bar, but with the wrong application window focused. This requires me to either switch to Godot again or to hide the incorrectly focused window.

**Steps to reproduce:**

1. Open another application
2. Open Mission Control
3. Select the Godot application

**Comment by akien-mga** [comment]
Is this still reproducible in the latest 3.2 builds?

**Comment by dogatekin** [comment]
This does happen to me even on the latest build. Godot is not the only application that does this on my computer, but it is one of a handful.

I'm guessing it's a macOS issue that happens for a group of applications to which Godot also belongs. I ran into this issue while searching for solutions and wanted to let you know. I'd be really happy if someone figured it out by any chance.

**Comment by tavurth** [comment]
This also happens to any exported Godot project

**Comment by tavurth** [comment]
Does anyone know what might be the entry point for digging into this issue?

I use Godot apps for quite a few custom built background apps now. When I swap apps it's regularly with three-fingers swipe-up to get to "Mission Control" then select the app.

This bug kills the ability for Godot to be used that way, I have to swipe up and then select two times.

If anyone has an idea of where to start looking, fixing this bug could save some frustration and allow a more native like eperience.

**Comment by afpro** [comment]
any update? facing exact same issue here.

**Comment by Calinou** [comment]
> any update? facing exact same issue here.

As far as I know, nobody knows a solution to this issue yet.

**Comment by ztc0611** [comment]
I experience this issue intermittently and it gets quite frustrating. Hopefully someday a fix is found. I attached a video in case that helps someone.

https://user-images.githubusercontent.com/28269330/186275130-9ffe2b0c-6eeb-4c77-bd36-ea85c05be7b9.mp4

**Comment by Calinou** [comment]
cc @bruvzg

**Comment by tbergeron** [comment]
Always happen to me when I CMD+TAB back into Godot, I **always** need to do it twice because the first time, it goes back to the last app that was on screen. It's similar to @ztc0611's video but with the main Godot window. It is so annoying and yet so hard to explain and/or debug...

**Comment by ztc0611** [comment]
@tbergeron Yeah its very weird for me, sometimes it does what you describe and sometimes it does what happens in my posted video. I've been trying to find a pattern of what causes it and as of yet haven't. Pico-8 has the same problem.

**Comment by afpro** [comment]
Not only the editor, even the 'play' window has this bug too.
Not sure whether exported mac app still has this strange behaviour or not.

**Comment by ztc0611** [comment]
Has anyone checked if this issue still occurs in the 4.0 alphas? It's not impossible that the refactors to how windows are handled accidentally fixed it. Might provide some inspiration of how to fix it for 3.x... or might show that a fix is basically impossible besides waiting for 4.

**Comment by ChrisAbra** [comment]
I'm still hitting this bug on 4.0 mono builds (havent tested regular yet)

**Comment by cridenour** [comment]
This definitely happens in 4.0 (from early betas to 4.0.2).

**Comment by ztc0611** [comment]
Still happens.
